src/pass/printer.cpp

```diff
@@ -268,10 +269,25 @@ void IRPrinter::Visit(Func& fn)
     for (auto& input : fn.inputs) { ostr << input->name << ", "; }
     ostr << (fn.inputs.size() > 0 ? "\b\b" : "") << ") {";
 
+    auto syminfo_map = SymAnalysis::Build(std::make_shared<Func>(fn));
+
+    std::vector<std::pair<Sym, SymInfo>> ordered_symbols;
+    for (const auto& [sym, info] : syminfo_map) {
+        ordered_symbols.emplace_back(sym, info);
+    }
+
+    std::sort(ordered_symbols.begin(), ordered_symbols.end(),
+              [](const auto& x, const auto& y) {
+                  return x.second.order < y.second.order;
+              });
+
     enter_block();
-    for (auto& [sym, val] : fn.tbl) {
-        emitassign(sym, val);
-        emitnewline();
+    for (const auto& [sym, info] : ordered_symbols) {
+        auto it = fn.tbl.find(sym);
+        if (it != fn.tbl.end()) {
```

Comment:
do we need this check? all the symbols in ordered_symbols should be in fn.tbl right?

Reply:
you can call fn.tbl.at(sym) to fetch the expression mapped to the sym

Reply:
I think it's necessary because I believe the input symbols are not in fn.tbl but they are included in the SymAnalysis pass (so thus they are in ordered_symbols). Since inputs printed out separately already as part of the function header, they can just be skipped here. 

I can add a comment to clarify that this if stmt is just to skip the input symbols if that'd be helpful.